server/src/main/java/org/apache/iotdb/db/metadata/idtable/IDTable.java

```diff
@@ -0,0 +1,181 @@
+   * @return reusable device id
+   * @throws MetadataException if insert plan's aligned value is inconsistent with device
+   */
+  IDeviceID getSeriesSchemas(InsertPlan plan) throws MetadataException;
+
+  /**
+   * update latest flushed time of one timeseries
+   *
+   * @param timeseriesID timeseries id
+   * @param flushTime latest flushed time
+   * @throws MetadataException throw if this timeseries is not exist
+   */
+  void updateLatestFlushTime(TimeseriesID timeseriesID, long flushTime) throws MetadataException;
+
+  /**
+   * update latest flushed time of one timeseries
+   *
+   * @param timeseriesID timeseries id
+   * @return latest flushed time of one timeseries
+   * @throws MetadataException throw if this timeseries is not exist
+   */
+  long getLatestFlushedTime(TimeseriesID timeseriesID) throws MetadataException;
+
+  /**
+   * register trigger to the timeseries
+   *
+   * @param fullPath full path of the timeseries
+   * @param measurementMNode the timeseries measurement mnode
+   * @throws MetadataException if the timeseries is not exits
+   */
+  void registerTrigger(PartialPath fullPath, IMeasurementMNode measurementMNode)
+      throws MetadataException;
+
+  /**
+   * deregister trigger to the timeseries
+   *
+   * @param fullPath full path of the timeseries
+   * @param measurementMNode the timeseries measurement mnode
+   * @throws MetadataException if the timeseries is not exits
+   */
+  void deregisterTrigger(PartialPath fullPath, IMeasurementMNode measurementMNode)
+      throws MetadataException;
+  /**
+   * get last cache of the timeseies
+   *
+   * @param timeseriesID timeseries ID of the timeseries
+   * @throws MetadataException if the timeseries is not exits
+   */
+  TimeValuePair getLastCache(TimeseriesID timeseriesID) throws MetadataException;
+
+  /**
+   * update last cache of the timeseies
+   *
+   * @param timeseriesID timeseries ID of the timeseries
+   * @param pair last time value pair
+   * @param highPriorityUpdate is high priority update
+   * @param latestFlushedTime last flushed time
+   * @throws MetadataException if the timeseries is not exits
+   */
+  void updateLastCache(
+      TimeseriesID timeseriesID,
+      TimeValuePair pair,
+      boolean highPriorityUpdate,
+      Long latestFlushedTime)
+      throws MetadataException;
+
+  /** clear id table and close file */
+  void clear();
+
+  /**
+   * translate query path's device path to device id
+   *
+   * @param fullPath full query path
+   * @return translated query path
+   */
+  static PartialPath translateQueryPath(PartialPath fullPath) {
+    // if not enable id table, just return original path
+    if (!config.isEnableIDTable()) {
+      return fullPath;
+    }
+
+    try {
+      // handle aligned path
+      if (fullPath instanceof AlignedPath) {
+        AlignedPath cur = (AlignedPath) fullPath;
+
+        return new AlignedPath(
+            DeviceIDFactory.getInstance().getDeviceID(cur).toStringID(),
```

Comment:
could we return the fullPath directly, just replace the path field.

Reply:
I think PartialPath is an immutable class and we can't replace filed of it. There is no method to change "nodes" filed of a partial path. I think immutable is reasonable.